src/@context/Pool/index.tsx

```diff
@@ -97,137 +89,120 @@ function PoolProvider({ children }: { children: ReactNode }): ReactElement {
-    calculateSharesVL(
-      poolData.id,
-      poolData.baseToken.address,
-      poolData.shares[0].shares,
-      asset.chainId
-    ).then((shares) => {
-      setOwnerPoolShares(shares)
-    })
-    // Total Liquidity
-    const totalLiquidityInOcean = new Decimal(
-      poolData.baseTokenLiquidity * 2 || 0
-    )
-
     const newPoolInfo = {
       liquidityProviderSwapFee: getFee(poolData.liquidityProviderSwapFee),
       publishMarketSwapFee: getFee(poolData.publishMarketSwapFee),
-      opcFee: getFee(poolData.opcFee),
+      opcFee: getFee(
+        getOpcForToken(poolData.baseToken.address, asset?.chainId)
+      ),
       weightBaseToken: getWeight(poolData.baseTokenWeight),
       weightDt: getWeight(poolData.datatokenWeight),
       datatokenSymbol: poolData.datatoken.symbol,
       datatokenAddress: poolData.datatoken.address,
       baseTokenSymbol: poolData.baseToken.symbol,
       baseTokenAddress: poolData.baseToken.address,
-      totalPoolTokens: poolData.totalShares,
-      totalLiquidityInOcean
+      totalPoolTokens: poolData.totalShares
     }
 
     setPoolInfo(newPoolInfo)
     LoggerInstance.log('[pool] Created new pool info:', newPoolInfo)
-  }, [asset?.chainId, chainId, poolData, web3])
+  }, [asset?.chainId, chainId, getOpcForToken, poolData, web3])
 
   //
   // 2 Pool Creator Info
   //
   useEffect(() => {
-    if (
-      !poolData ||
-      !poolInfo?.totalPoolTokens ||
-      !poolInfo.totalLiquidityInOcean ||
-      ownerPoolShares === '0'
-    )
-      return
-
-    // Pool share tokens.
-    const poolShare = new Decimal(ownerPoolShares)
-      .dividedBy(poolInfo.totalLiquidityInOcean)
-      .mul(100)
-      .toFixed(2)
-
-    const newPoolOwnerInfo = {
-      liquidity: new Decimal(ownerPoolShares), // liquidity in base token, values from from `calcSingleOutGivenPoolIn` method
-      poolShares: ownerPoolShares,
-      poolShare
+    async function calculatePoolOwnerInfo() {
+      if (
+        !poolData ||
+        !poolInfo?.totalPoolTokens ||
+        poolData.shares[0]?.shares === '0'
+      )
+        return
+
+      // Pool share tokens.
+      const poolSharePercentage = new Decimal(poolData.shares[0]?.shares)
+        .dividedBy(poolInfo.totalPoolTokens)
+        .mul(100)
+        .toFixed(2)
+
+      const ownerLiquidity = calcSingleOutGivenPoolIn(
+        poolData.baseTokenLiquidity,
+        poolData.totalShares,
+        poolData?.shares[0]?.shares
+      )
+
+      const newPoolOwnerInfo = {
+        liquidity: ownerLiquidity,
+        poolShares: poolData.shares[0]?.shares,
+        poolSharePercentage
+      }
+      setPoolInfoOwner(newPoolOwnerInfo)
+      LoggerInstance.log(
+        '[pool] Created new pool creatorinfo:',
+        newPoolOwnerInfo
+      )
     }
-    setPoolInfoOwner(newPoolOwnerInfo)
-    LoggerInstance.log('[pool] Created new owner pool info:', newPoolOwnerInfo)
+
+    calculatePoolOwnerInfo()
   }, [
-    ownerPoolShares,
+    asset.chainId,
```

Comment:
throughout this file we seem to assume that `asset` can be undefined as we are doing `asset?....` a lot so I guess we should keep it for every `asset` access in this file, so in this case `asset?.chainId`